### PR TITLE
Update READMEs

### PR DESCRIPTION
**Pull request by dra27**
Done in ~~4 commits - in particular, the "5.0 cautionary note" might want to remain for now.~~ 3 commits - cf. #13394 for the cationary note.

### CODE REVIEW

File: README.adoc [hunk 1]
```diff
@@ -83,6 +60,10 @@ compiler currently runs on the following platforms:
 Other operating systems for the processors above have not been tested, but
 the compiler may work under other operating systems with little work.
 
+❗ From OCaml 5.0 onwards, native compilation is available only on 64-bit
```
Comment (ghost) [inline README.adoc:80]: Maybe
```suggestion
❗ From OCaml 5.0 onwards, native compilation is available only on some 64-bit
```
or (`a few`)

**Review by ghost — APPROVED** [review]

**Review by shindere — APPROVED** [review]

**Comment by shindere** [comment]
A very cool update, thanks!

File: README.adoc (hunk 1 above)
Comment (dra27) [reply README.adoc:80]: That makes it sound as though it's fewer 64-bit systems from OCaml 5.0, though? Or perhaps drop the "From OCaml 5.0 onwards," as well?

File: README.adoc [hunk 2]
```diff
@@ -82,6 +77,10 @@ compiler currently runs on the following platforms:
...
+systems.  Native compilation on 32-bit systems is no longer available, nor
+are there plans to bring it back.
+all architectures.
```
Comment (jmid) [inline README.adoc]:
```suggestion

```

I think this was a copy-paste left over.

Comment (dra27) [reply README.adoc]: Oo indeed - or not enough `dd` 🙂

**Comment by gasche** [comment]
I have mixed feelings about removing the cautionary note. The new runtime is still green in my eyes, and we suffer from known GC pacing issues (and probably some other unknown issues) that result in real regressions for existing programs, whose users are better off with 4.14. (Coq is one of them, Liquidsoap; Frama-C looks like it is now at performance parity but this is not fully clear, etc.) And for now we are still considering the Domain interface experimental, and changing things in language support for effects, etc. I think that a word of warning that our 5.x offering is not quite as production-ready as 4.14 for existing software remains appropriate.

**Comment by dra27** [comment]
OK - I certainly think we should soften the warning, but I've removed that commit and will open that in a separate PR so as not to hold up the rest.

File: README.win32.adoc [hunk 3]
```diff
@@ -100,22 +100,18 @@ The native-code compiler (`ocamlopt`) and static linking of OCaml bytecode with
...
+The OCaml runtime requires C11 atomics, which are available in
+Visual Studio 2022 (starting from release 17.8). Earlier versions (either of
+Visual Studio 2022 17.7 and earlier or Visual Studio 2019 and earlier) cannot
+be used to compile OCaml. In addition to the paid versions of Visual Studio,
+OCaml can be compiled with either Visual Studio 2022 Community or Build Tools
+for Visual Studio 2022. The following components are needed (the precise version
+numbers may differ, but this should not matter):
+
+ - MSVC v143 - VS 2022 C++ x64/x86 build tools (Latest)
+ - C\++/CLI support for v143 build tools (Latest)
```
Comment (MisterDA) [inline README.win32.adoc]:
```suggestion
 - C++/CLI support for v143 build tools (Latest)
```

I suppose this was copied from asciidoc.

**Review by MisterDA — APPROVED** [review]
Maybe a short mention that clang-cl may also be used?

**Comment by shindere** [comment]
Antonin Décimo (2024/08/23 04:59 -0700):
> Maybe a short mention that clang-cl may also be used?

Yeah, I did have that in mind, too!

File: README.win32.adoc (hunk 3 above)
Comment (dra27) [reply README.win32.adoc]: It is asciidoc? But it also wasn't rendering correctly - fixed, thanks!

**Comment by dra27** [comment]
How's the last commit look, @MisterDA - I didn't double-check the instructions?

**Review by MisterDA — APPROVED** [review]
> How's the last commit look, @MisterDA - I didn't double-check the instructions?

Exactly as it should be. We could even remove the `--build` part of configure, right?

**Comment by MisterDA** [comment]
While we're at it, I'd put the "for 32-bit" and "for 64-bit" _before_ the `configure` command, and use Markdown's backticks with `sh` for the code snippets:

> Now run for 32-bit:
> ```sh
> ./configure --build=x86_64-pc-cygwin --host=i686-pc-windows
> ```
> or for 64-bit:
> ```sh
> ./configure --build=x86_64-pc-cygwin --host=x86_64-pc-windows
> ```

(sorry for the back and forth).

**Comment by shindere** [comment]
David Allsopp (2024/08/23 05:59 -0700):
> It is asciidoc? But it also wasn't rendering correctly - fixed,
> thanks!

Speaking about that (not for this PR though): do we really want to
continue dealing with several file formats for ourdocumentation? At
least for the README files, don't we want to be all markdown?

**Comment by shindere** [comment]
Antonin Décimo (2024/08/23 06:03 -0700):
> We could even remove the `--build` part of configure, right?

I think it's an autoconf requirement that if you specify --hsot then
it's recommended to also specify --build

>
>
>

**Comment by shindere** [comment]
Yeah that's something I long wanted to see changed that the `for this`
bit comes before the associated instruction rather than after it. Well
spotted @MisterDA, thanks!

**Comment by dra27** [comment]
FWIW, I don't find "being all markdown" an interesting end/goal on its own. AsciiDoc was picked with good reasons in #242, but that was nearly 9 years ago, so if we can now do a good enough job with Markdown alone then it could maybe be revisited (but not by me!).

**Comment by dra27** [comment]
How's that one look? I also put the 64-bit version first, given that's what we really should be recommending these days

**Review by MisterDA — APPROVED** [review]

**Comment by dra27** [comment]
Thank you all for the reviews and suggestions!

**Comment by MisterDA** [comment]
> if we can now do a good enough job with Markdown alone then it could maybe be revisited (but not by me!).

Oh I think I had a brain reboot at some point and I thought we were editing a markdown document.

**Pull request merged**